lib/signaling/v2/localtrackpublication.js

```diff
@@ -15,6 +16,12 @@ class LocalTrackPublicationV2 extends LocalTrackPublicationSignaling {
    */
   constructor(trackSender, name, priority) {
     super(trackSender, name, priority);
+    Object.defineProperties(this, {
+      _mediaStates: {
```

Comment:
suggest rename this to `recordingWarningState`

Reply:
I think mediaStates make more sense as this will hold all media states (not just warnings) for both recorded and non recorded media. This mirrors the RSP message in the blueprint.